addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -222,6 +222,13 @@ options.registry.gallery = options.Class.extend({
         this.$target
             .removeClass('o_nomode o_masonry o_grid o_slideshow')
             .addClass('o_' + widgetValue);
+        // Used to prevent the editor's "unbreakable protection mechanism" from
+        // restoring Image Wall adaptations (images removed > new images added
+        // to the container & layout updates) when adding new images to the
+        // snippet.
+        if (this.options.wysiwyg) {
+            this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
+        }
```

Comment:
I guess we can merge this to solve the OPW asap but this seems like a band-aid on the problem, I don't understand why the unbreakable system breaks things here :grimacing: Let's merge but do you have a better explanation? If not it may be worth notifying to the editor team :grimacing:

Reply:
@qsm-odoo Yes, as I explained on discord, I couldn't find, while debugging, any suspicious code that triggers this behaviour (the code on "OdooEditor" will simply rollback the mutations of a container considered as unbreakable by checking nodes ouid...). By doing a quick check, I can see that we already had the same [issue & fix in the past](https://github.com/odoo/odoo/pull/69156) and from the discussion on the PR, it seems that "unbreakable protection mechanism" was implemented for some specific situations, but it affects the snippet changes too... :thinking: so Yes we can get more details from the editor team about this part of the code, maybe this conflict can be prevented in a global way :+1:

Reply:
@dmo-odoo As explained by @xO-Tx I already r+'ed such a diff in the past without really liking it. There is either something to fix in a more global way or something that I don't understand, maybe worth having a look? :)

Summary: "We apparently need to call `unbreakableStepUnactive` in some places to make it work but we don't see why the unbreakable rollback mechanism is even triggered in those places". Notice that here, the problem was also only on Chrome (not Firefox) so this is even stranger to me.